hashing/hashing.go

```diff
@@ -0,0 +1,57 @@
+package hashing
+
+import (
+	"fmt"
+	"hash/fnv"
+	"k8s.io/apimachinery/pkg/util/rand"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+	"reflect"
+)
+
+func ComputeHashForKubernetesObject(object interface{}) (string, error) {
+	value := reflect.ValueOf(object)
+	if value.Kind() != reflect.Ptr {
+		return "", fmt.Errorf("Must pass pointer to ComputeHashForKubernetesObject!")
+		// This is because the hash function prints the type which will be different
+		// if the object is a pointer (even if they are otherwise identical)
+		// normally we would only accept a pointer but this has to accept an interface
+		// so we can use it with any k8s object.
+	}
+	objectMeta := value.Elem().FieldByName("ObjectMeta")
+	if objectMeta.Kind() == reflect.Struct {
+		labels := objectMeta.FieldByName("Labels")
+		if labels.Kind() == reflect.Map {
+			// equivalent of delete(labels, "yelp.com/operator_config_hash")
+			// we remove this so that the already hashed versions will match
+			// the newly calculated versions
+			labels.SetMapIndex(reflect.ValueOf("yelp.com/operator_config_hash"), reflect.Value{})
+		}
+	}
+	hasher := fnv.New32a()
+	hashutil.DeepHashObject(hasher, object)
```

Comment:
this could be too hair-triggery. we  should consider deephashing only `.ObjectMeta.Labels` and `.Spec` if those exist in a given object?

Reply:
Yep, I kinda agree. I'm not fully sure though. It'd be cool if this thing could take a list of things to include/not include like we do in paasta. That might be kinda tricky with golang and reflection though...

In general at the moment I think if anything has changed I'm okay with updating the object in k8s but it maybe that each operator will need to be more nuanced. Is there some field in-particular you are worried about?

Reply:
Oh yes `.Status` is definitely a problem. Okay I'm convinced let me change this to just labels+spec.